### PR TITLE
fix: issue with did resolution

### PR DESCRIPTION
**Pull request by maycon-mello**

### CODE REVIEW

File: src/features/didManagement/didAuthHooks.js [hunk 1]
```diff
@@ -54,18 +51,20 @@ export function useDIDAuth() {
...
-  const getSelectedDIDKeyDoc = useCallback(
-    async ({selectedDID}) => {
-      console.log(keyDocs, selectedDID);
-      const keyDoc = keyDocs.find(doc => doc.controller === selectedDID);
+  const getSelectedDIDKeyDoc = useCallback(async ({selectedDID}) => {
+    const {didDocument} = await walletService.getDocumentById(selectedDID);
+    const allDocs = await walletService.query({});
+    const keyDoc = allDocs.find(
+      document =>
+        document?.id.indexOf(didDocument.id) > -1 &&
```
Comment (mike-parkhill) [inline src/features/didManagement/didAuthHooks.js:59]: could `document?.id` or `document?.type` be undefined?

File: src/features/didManagement/didAuthHooks.js [hunk 2]
```diff
@@ -54,18 +51,20 @@ export function useDIDAuth() {
...
+        document?.type.indexOf('VerificationKey') > -1,
+    );
+    if (keyDoc) {
+      return keyDoc;
+    }
+    throw new Error('DID KeyDocument not found');
```
Comment (mike-parkhill) [inline src/features/didManagement/didAuthHooks.js:65]: is this error being handled gracefully?

File: test/testSetup.js [hunk 3]
```diff
@@ -421,40 +421,49 @@ jest.mock('@docknetwork/wallet-sdk-core/lib/services/wallet', () => {
...
+        'did:key:z6MkjjCpsoQrwnEmqHzLdxWowXk5gjbwor4urC1RPDmGeV8r',
+      type: 'Ed25519VerificationKey2018',
+      privateKeyBase58:
+        '3CQCBKF3Mf1tU5q1FLpHpbxYrNYxLiZk4adDtfyPEfc39Wk6gsTb2qoc1ZtpqzJYdM1rG4gpaD3ZVKdkiDrkLF1p',
+      publicKeyBase58: '6GwnHZARcEkJio9dxPYy6SC5sAL6PxpZAB6VYwoFjGMU',
+    },
+  ];
+
   const mockFunctions = {
+    query: jest.fn(() => {
```
[the remark this answers is not in the document]
Comment (mike-parkhill) [reply test/testSetup.js:459]: what do you need for that to happen? Just some time to focus on it?

[the remark this answers is not in the document]
Comment (maycon-mello) [reply test/testSetup.js:459]: yes, I can create a ticket to work on it. Will need to review all the unit tests using mocks for wallet-sdk functions. We can handle this in multiple PRs

[the remark this answers is not in the document]
Comment (mike-parkhill) [reply test/testSetup.js:459]: might make sense to create one ticket to figure out a plan for migrating them and then having a bunch of small tickets for actually doing it

**Review by mike-parkhill — APPROVED** [review]

**Review by SamHellawell — APPROVED** [review]

**Pull request merged**